include/NeoFOAM/fields/comparisons/FieldComparison.hpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include "NeoFOAM/fields/Field.hpp"
+#include <span>
+
+namespace NeoFOAM
+{
+
+template<typename T>
+bool compare(Field<T>& field, T value)
+{
+    auto hostSpan = field.copyToHost().field();
+    for (int i = 0; i < hostSpan.size(); i++)
+    {
+        if (hostSpan[i] != value)
+        {
+            return false;
+        }
+    }
+    return true;
+};
+
+template<typename T>
+bool compare(Field<T>& field, Field<T>& field2)
```

Comment:
Shouldn't they be const?
```suggestion
bool compare(const Field<T>& field, const Field<T>& field2)
```